RUNTIME.md

```diff
@@ -0,0 +1,536 @@
+"mpiexec" where the wrapper script enables the GEOPM runtime.  The
+"geopmlaunch" command supports exactly the same command line interface
+as the underlying launch command, but the wrapper extends the
+interface with GEOPM specific options.  The "geopmlaunch" application
+launches the primary compute application and the GEOPM control thread
+on each compute node and manages the CPU affinity requirements for all
+processes.  The wrapper is documented in the geopmlaunch(1) man page.
+
+There are several underlying MPI application launchers that
+"geopmlaunch" wrapper supports.  See the geopmlaunch(1) man page for
+information on available launchers and how to select them.  If the
+launch mechanism for your system is not supported, then affinity
+requirements must be enforced by the user and all options to the GEOPM
+runtime must be passed through environment variables.  Please consult
+the geopm(7) man page for documentation of the environment variables
+used by the GEOPM runtime that are otherwise controlled by the wrapper
+script.
+
+### CPU AFFINITY REQUIREMENTS
+The GEOPM runtime requires that each MPI process of the application
+under control is affinitized to distinct CPUs.  This is a strict
+requirement for the runtime and must be enforced by the MPI launch
+command.  When using the geopmlaunch wrapper described in the previous
+section, these affinity requirements are handled by geopmlaunch unless
+the --geopm-affinity-disable command line option is provided (see
+geopmlaunch(1)).
+
+While the GEOPM control thread connects to the application it will
+automatically affinitize itself to the highest indexed core not used
+by the application if the application is not affinitized to a CPU on
+every core.  In the case were the application is utilizing all cores
+of the system, the GEOPM control thread will be pinned to the highest
+logical CPU.
+
+There are many ways to launch an MPI application, and there is no
+single uniform way of enforcing MPI rank CPU affinities across
+different job launch mechanisms.  Additionally, OpenMP runtimes, which
+are associated with the compiler choice, have different mechanisms for
+affinitizing OpenMP threads within CPUs available to each MPI process.
+To complicate things further the GEOPM control thread can be launched
+as an application thread or a process that may be part of the primary
+MPI application or a completely separate MPI application.  For these
+reasons it is difficult to document how to correctly affinitize
+processes in all configurations.  Please refer to your site
+documentation about CPU affinity for the best solution on the system
+you are using and consider extending the geopmlaunch wrapper to
+support your system configuration (please see the CONTRIBUTING.md file
+for information about how to share these implementation with the
+community).
+
+TESTING
+-------
+From within the source code directory, unit tests can be executed with
```

Comment:
Which source code directory?  The base or service?

We should be explicit about their being two different spots that you need to do this process in.  Examples would be nice too.